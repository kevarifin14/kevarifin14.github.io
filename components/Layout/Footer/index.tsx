import { FaLinkedinIn, FaTwitter } from 'react-icons/fa';

import FooterLink from './FooterLink';

export default function Footer() {
  const links = [
    { Icon: FaTwitter, href: 'https://twitter.com/kevarifin' },
    { Icon: FaLinkedinIn, href: 'https://www.linkedin.com/in/kevinarifin/' },
  ];

  return (
    <div className="border-t-2 border-gray-100 py-8 max-w-7xl w-full mx-auto">
      <div className="flex justify-center flex-col items-center space-y-4 md:space-y-0 md:flex-row md:justify-between">

        <div className="flex space-x-6 md:order-2">
          {links.map(({ Icon, href }) => <FooterLink Icon={Icon} href={href} />)}
        </div>

        <p className="text-base text-gray-400 xl:text-center md:order-1">
          &copy; 2021 Kevin Arifin
        </p>

      </div>
    </div>
  );
}
