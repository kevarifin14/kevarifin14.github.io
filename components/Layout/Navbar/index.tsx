import { Transition } from '@headlessui/react';
import { useRouter } from 'next/router';
import { useState } from 'react';
import {
  HiBookOpen, HiDocumentText, HiMail, HiUser,
} from 'react-icons/hi';

import { ILink } from './NavLink';
import NavLogo from './NavLogo';
import NavMenu from './NavMenu';
import NavMobileMenu from './NavMobileMenu';
import NavMenuButton from './NavOpenButton';

export default function Navbar() {
  const router = useRouter();
  const [isVisible, setIsVisible] = useState(false);

  const links: ILink[] = [
    {
      title: 'Blog',
      href: '/blog',
      active: router.pathname.includes('/blog'),
      icon: HiDocumentText,
    },
    {
      title: 'Newsletter',
      href: '/newsletter',
      active: router.pathname.includes('/newsletter'),
      icon: HiMail,
    },
    {
      title: 'About',
      href: '/about',
      active: router.pathname.includes('/about'),
      icon: HiUser,
    },
  ];

  return (
    <div className="relative max-w-7xl mx-auto w-full px-4 sm:px-6 lg:px-8">
      <div className="flex justify-between items-center pt-4 md:space-x-10">

        <NavLogo />

        <div className="-mr-2 -my-2 md:hidden">
          <NavMenuButton onClick={() => setIsVisible(true)} />
        </div>

        <NavMenu className="hidden md:flex" links={links} />

      </div>

      <Transition
        show={isVisible}
        enter="duration-200 ease-out"
        enterFrom="opacity-0 scale-95"
        enterTo="opacity-100 scale-100"
        leave="duration-100 ease-in"
        leaveFrom="opacity-100 scale-100"
        leaveTo="opacity-0 scale-95"
        className="absolute top-0 inset-x-0 py-2 transition transform origin-top-right md:hidden z-50"
      >
        <NavMobileMenu
          links={links}
          onClick={() => setIsVisible(false)}
        />
      </Transition>
    </div>
  );
}
