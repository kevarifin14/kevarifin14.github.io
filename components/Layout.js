import Head from 'next/head'

import Navbar from './Navbar';

export default function Layout({ children, showLogo, title }) {
  return (
    <div className="container">
      <Head>
        <title>{`${title} | Kevin Arifin`}</title>

        <link rel="icon" href="/favicon.png" />
        <link
          href="https://fonts.googleapis.com/css2?family=Open+Sans&display=swap"
          rel="stylesheet"
        />

        <meta name="twitter:card" content="https://kevinarifin.com/favicon.png" key="twcard" />
        <meta name="twitter:creator" content="kevarifin" key="twhandle" />

        <meta property="og:url" content="kevinarifin.com" key="ogurl" />
        <meta property="og:image" content="https://kevinarifin.com/favicon.png" key="ogimage" />
        <meta property="og:site_name" content="Kevin Arifin" key="ogsitename" />
        <meta property="og:title" content={`${title} | Kevin Arifin`} key="ogtitle" />
        <meta property="og:description" content="Kevin's Personal Website" key="ogdesc" />
      </Head>

      <Navbar showLogo={showLogo} />

      {children}

      <footer>
        <p>Copyright Â© 2020 Kevin Arifin</p>
      </footer>

      <style jsx>{`
        .container {
          min-height: 100vh;
          padding: 0 0.5rem;
          display: flex;
          flex-direction: column;
        }

        footer {
          width: 100%;
          height: 100px;
          border-top: 1px solid #eaeaea;
          display: flex;
          justify-content: center;
          align-items: center;
        }

      `}</style>

      <style jsx global>{`
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: Open Sans,-apple-system, BlinkMacSystemFont, Segoe UI, Roboto,
            Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue,
            sans-serif;
        }

        * {
          box-sizing: border-box;
        }

        hr {
          background-color: #eaeaea;
          height: 2px;
          border-width: 0;
        }
      `}</style>
    </div>
  );
}