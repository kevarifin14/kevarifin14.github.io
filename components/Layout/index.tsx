import Head from 'next/head';
import { ReactNode } from 'react';

import NotificationOverlay from 'components/NotificationOverlay';
import NotificationProvider from 'components/NotificationProvider';

import Footer from './Footer';
import Navbar from './Navbar';

type LandingPageLayoutProps = {
  children: ReactNode,
};

export default function Layout({ children }: LandingPageLayoutProps) {
  return (
    <NotificationProvider>

      <NotificationOverlay />

      <div>
        <Head>
          <title>Kevin Arifin</title>
        </Head>

        <div className="flex flex-col px-4 sm:px-6 lg:px-8">

          <main className="min-h-screen max-w-7xl w-full mx-auto">
            <Navbar />
            {children}
          </main>

          <Footer />

        </div>

      </div>
    </NotificationProvider>
  );
}

export const getLayout = (page) => <Layout>{page}</Layout>;
