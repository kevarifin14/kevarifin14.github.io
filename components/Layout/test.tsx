import renderer from 'react-test-renderer';

import Layout from '.';

describe('Layout', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <Layout />,
    );
    expect(tree).toMatchSnapshot();
  });
});
