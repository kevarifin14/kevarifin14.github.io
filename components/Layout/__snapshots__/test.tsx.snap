// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
Array [
  <div
    className="fixed max-w-sm w-full right-0 top-0 flex flex-col justify-end items-start space-y-4 pointer-events-none z-30 p-4"
  />,
  <div
    className="bg-light"
  >
    <div
      className="min-h-screen flex flex-col"
    >
      <main
        className="max-w-7xl w-full mx-auto px-4 sm:px-6 lg:px-8"
      >
        <div
          className="relative max-w-7xl mx-auto w-full"
        >
          <div>
            <div
              className="flex justify-between items-center border-b-2 border-gray-100 py-4 md:space-x-10"
            >
              <div
                className="relative h-16 w-16 cursor-pointer"
              >
                <div
                  style={
                    Object {
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "left": 0,
                      "margin": 0,
                      "overflow": "hidden",
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <img
                    decoding="async"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style={
                      Object {
                        "border": "none",
                        "bottom": 0,
                        "boxSizing": "border-box",
                        "display": "block",
                        "height": 0,
                        "left": 0,
                        "margin": "auto",
                        "maxHeight": "100%",
                        "maxWidth": "100%",
                        "minHeight": "100%",
                        "minWidth": "100%",
                        "objectFit": "contain",
                        "objectPosition": undefined,
                        "padding": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "visibility": "hidden",
                        "width": 0,
                      }
                    }
                  />
                </div>
              </div>
              <div
                className="-mr-2 -my-2 md:hidden"
              >
                <button
                  className="bg-white rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="h-6 w-6"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4 6h16M4 12h16M4 18h16"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                    />
                  </svg>
                </button>
              </div>
              <nav
                className="space-x-10 hidden md:flex"
              >
                <a
                  className="text-lg font-medium text-gray-500 hover:text-gray-900 no-underline"
                  href="/blog"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Blog
                </a>
                <a
                  className="text-lg font-medium text-gray-500 hover:text-gray-900 no-underline"
                  href="/tb"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  Newsletter
                </a>
                <a
                  className="text-lg font-medium text-gray-500 hover:text-gray-900 no-underline"
                  href="/about"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  About
                </a>
              </nav>
            </div>
          </div>
        </div>
      </main>
    </div>
    <div
      className="border-t-2 border-gray-100 py-8 max-w-7xl w-full mx-auto"
    >
      <div
        className="flex justify-center flex-col items-center space-y-4 md:space-y-0 md:flex-row md:justify-between"
      >
        <div
          className="flex space-x-6 md:order-2"
        >
          <a
            className="text-gray-400 hover:text-gray-500"
            href="https://twitter.com/kevarifin"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              className="h-6 w-6"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
          </a>
          <a
            className="text-gray-400 hover:text-gray-500"
            href="https://www.linkedin.com/in/kevinarifin/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              className="h-6 w-6"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
              />
            </svg>
          </a>
        </div>
        <p
          className="text-base text-gray-400 xl:text-center md:order-1"
        >
          Â© 2021 Kevin Arifin
        </p>
      </div>
    </div>
  </div>,
]
`;
